x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useState, useMemo, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  EuiFieldText,
+  EuiText,
+  EuiIconTip,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { cloneDeep } from 'lodash';
+import { policyConfig } from '../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from './policy_hooks';
+
+function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
+  let newPolicyConfig = obj;
+  for (let i = 0; i < path.length - 1; i++) {
+    newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
+  }
+  newPolicyConfig[path[path.length - 1]] = value;
+}
+
+function getValue(obj: Record<string, unknown>, path: string[]) {
```

Comment:
and just make sure that we only get the value if it exists, otherwise we can just return `undefined`.  again, code can be cleaned up, just communicating an idea

```js
function getValue(obj: Record<string, unknown>, path: string[]) {
  let currentPolicyConfig = obj;

  for (let i = 0; i < path.length - 1; i++) {
    if(currentPolicyConfig[path[i]]) {
      currentPolicyConfig = currentPolicyConfig[path[i]] as Record<string, unknown>;
    } else {
      return undefined;
    }
  }
  return currentPolicyConfig[path[path.length - 1]];
}
```

Reply:
is having multiple `return`s common practice on this team?